portfolio/src/main/java/com/google/sps/servlets/BlogCommentServlet.java

```diff
@@ -65,64 +40,62 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String sort = request.getParameter("sort");
 
     // get each result from datastore and generate comments 
-    List<Entity> results = service.findAllComments(num, id, sort, true);
-    ArrayList<Comment> comments = new ArrayList<Comment>();
-    for (Entity entity : results) {
-      String content = (String) entity.getProperty("content");
-      Date time = (Date) entity.getProperty("time");
-      String name = (String) entity.getProperty("name");
-      long postId = (Long) entity.getProperty("postid");
-      long commentId = (Long) entity.getKey().getId();
-      String emoji = (String) entity.getProperty("emoji");
-      String email = (String) entity.getProperty("email");
-      String image = (String) entity.getProperty("image");
-      double score = (Double) entity.getProperty("score");
-      String classification = (String) entity.getProperty("classification");
-      
-      Comment comment = new Comment(content, time, name, postId, commentId, emoji, email, image, score, classification);
-      comments.add(comment);
-    }
-
+    ArrayList<Comment> comments = service.findAllComments(num, id, sort, true);
+  
     // Send the JSON as the response
-    String json = convertToJson(comments);
+    String json = service.convertToJson(comments);
     response.setContentType("application/json; charset=utf-8");
     response.getWriter().println(json);
   }
  
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // get parameters and create a new entity
-    UserService userService = UserServiceFactory.getUserService();
+    String action = request.getParameter("action");
+    if (action == null || !action.equals("delete")) {
```

Comment:
To improve readability, you might consider creating a private method for `addComment` and `deleteComment`. Otherwise, this `doPost` method is kind of long and it's not immediately clear that the default action is `create`.